pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+type PodSandboxID string
+
+// PodSandboxManager provides basic operations to create/delete and examine the
+// PodSandboxes.
+type PodSandboxManager interface {
+	// Create creates a sandbox based on the given config, and returns the
+	//  of the new sandbox.
+	Create(config *PodSandboxConfig) (PodSandboxID, error)
+	// Delete deletes the sandbox by its ID. If there are any running
+	// containers in the sandbox, they will be terminated as a side-effect.
+	Delete(id PodSandboxID) error
+	// List lists existing sandboxes, filtered by the given PodSandboxFilter.
+	List(filter PodSandboxFilter) []PodSandboxListItem
+	// Status gets the status of the sandbox by ID.
+	Status(id PodSandboxID) (PodSandboxStatus, error)
+	// PortForward copies the data from the port in the PodSandBox to the
+	// stream.
+	// TODO: Should PortForward be in a separate interface.
+	PortForward(id PodSandboxID, port uint16, stream io.ReadWriteCloser) error
```

Comment:
@pskrzyns I don't think it possible to do so. Even though, it will not work if isolated network is enabled in [network policy](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/network-policy.md).